src/components/HistoryTable/HistoryTableContext.tsx

```diff
@@ -0,0 +1,92 @@
+import { useContext, createContext, useReducer, useMemo } from "react";
+import { reducer } from "./historyTableContextReducer";
+import { rowType, mainlineCommits, CommitRowType } from "./utils";
+
+interface HistoryTableState {
+  itemHeight: (index: number) => number;
+  fetchNewCommit: (data: mainlineCommits) => void;
+  isItemLoaded: (index: number) => boolean;
+  getItem: (index: number) => CommitRowType;
+  processedCommitCount: number;
+  processedCommits: CommitRowType[];
+  visibleColumns: string[];
+  addColumns: (columns: string[]) => void;
+  nextPage: () => void;
+  previousPage: () => void;
+}
+
+const HistoryTableDispatchContext = createContext<any | null>(null);
+
+const HistoryTableProvider: React.FC = ({ children }) => {
+  const [
+    { processedCommits, processedCommitCount, visibleColumns },
+    dispatch,
+  ] = useReducer(reducer, {
+    loadedCommits: [],
+    processedCommits: [],
+    processedCommitCount: 0,
+    commitCache: {},
```

Comment:
This is probably a good use case for a map! The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#objects_vs._maps) suggest maps could yield some performance gains:

> Performs better in scenarios involving frequent additions and removals of key-value pairs.

This would perhaps also eliminate the need for the `cacheSize` field because of its `size` property.